### PR TITLE
[TD] Implements Coord related functions in C++.

### PR DESCRIPTION
**Pull request by OmniBlade**
Functions are based on the RA versions which are official translations of the assembly.

### CODE REVIEW

**Review by hifi — APPROVED** [review]
Indeed. These are direct ports of the RA versions and I see no ill effects in-game so testing these is not strictly necessary.

File: tiberiandawn/coord.cpp [hunk 1]
```diff
@@ -43,6 +43,63 @@
...
+int Distance_Coord(COORDINATE coord1, COORDINATE coord2)
+{
+    int diff1, diff2;
+
+    diff1 = Coord_Y(coord1) - Coord_Y(coord2);
+    if (diff1 < 0)
+        diff1 = -diff1;
+    diff2 = Coord_X(coord1) - Coord_X(coord2);
+    if (diff2 < 0)
+        diff2 = -diff2;
```
Comment (ghost) [inline tiberiandawn/coord.cpp:94]: maybe just use of abs on subtraction? should be a bit easier to read

File: tiberiandawn/coord.cpp [hunk 2]
```diff
@@ -43,6 +43,63 @@
...
+ * OUTPUT:  Returns the lepton distance between the two coordinates.                           *
+ *                                                                                             *
+ * WARNINGS:   none                                                                            *
+ *                                                                                             *
+ * HISTORY:                                                                                    *
+ *   05/27/1994 JLB : Created.                                                                 *
+ *=============================================================================================*/
+int Distance_Coord(COORDINATE coord1, COORDINATE coord2)
+{
+    int diff1, diff2;
```
Comment (ghost) [inline tiberiandawn/coord.cpp:87]: maybe `diff_X` and `diff_Y`?

**Review by ghost — APPROVED** [review]

File: tiberiandawn/coord.cpp (hunk 2 above)
Comment (OmniBlade) [reply tiberiandawn/coord.cpp:87]: Again a bit out of scope for this PR. The coord stuff is prime for merging between the two games, and this sort of thing will fall into that, but it requires work to increase the internal map size for TD to 128x128 to match RA.

**Pull request merged**